cmd/flags.go

```diff
@@ -67,24 +75,64 @@ func ISPProvider() isp.Provider {
 
 	return ispProvider
 }
-
+func parseRedisURL(redisURL string) (password string, hosts []string, err error) {
+	matches := redisURLRegExp.FindStringSubmatch(redisURL)
+	if len(matches) < 3 {
+		return "", nil, fmt.Errorf("should match %s", redisURLRegExp.String())
+	}
+	return matches[1], strings.Split(matches[2], ","), nil
+}
 func RedisClient() *redis.Client {
 	if *RedisAddr == "" {
-		log.Debug("Redis not configured")
+		log.Error("Redis not configured")
 		return nil
 	}
+	if _, err := os.Stat(*RedisCA); os.IsNotExist(err) {
+		log.Fatalf("Cannot find certificate authority file: %v", *RedisCA)
+	}
+	if _, err := os.Stat(*RedisClientPK); os.IsNotExist(err) {
+		log.Fatalf("Cannot find client private key file: %v", *RedisClientPK)
+	}
+	if _, err := os.Stat(*RedisClientCert); os.IsNotExist(err) {
+		log.Fatalf("Cannot find client certificate file: %v", *RedisClientCert)
+	}
+	redisClientCert, err := tls.LoadX509KeyPair(*RedisClientCert, *RedisClientPK)
+	if err != nil {
+		log.Fatalf("Failed to load client certificate: %v", err)
+	}
+	redisCACert, err := keyman.LoadCertificateFromFile(*RedisCA)
+	if err != nil {
+		log.Fatalf("Failed to load CA cert: %v", err)
+	}
+
+	redisPassword, redisHosts, err := parseRedisURL(*RedisAddr)
+	if err != nil {
+		log.Fatalf("Failed to parse Redis URL: %v", err)
+	}
+
 	log.Debugf("Connecting to Redis at %v", *RedisAddr)
-	redisClient, err := tlsredis.GetClient(&tlsredis.Options{
-		RedisURL:       *RedisAddr,
-		RedisCAFile:    *RedisCA,
-		ClientPKFile:   *RedisClientPK,
-		ClientCertFile: *RedisClientCert,
-	})
-	if err == nil {
-		log.Debugf("Connected to Redis at %v", *RedisAddr)
-		return redisClient
-	} else {
-		log.Errorf("Unable to connect to redis: %v", err)
+	redisOpts := redis.FailoverOptions{
+		SentinelAddrs:    redisHosts,
+		SentinelPassword: redisPassword,
+		Password:         redisPassword,
+		MasterName:       "mymaster",
+		// We use TLS as the transport. If we simply specify the TLSConfig, the Redis library will
+		// establish TLS connections to Sentinel, but plain TCP connections to masters.
+		Dialer: func(ctx context.Context, network, addr string) (net.Conn, error) {
+			return tls.Dial(network, addr, &tls.Config{
+				RootCAs:            redisCACert.PoolContainingCert(),
+				Certificates:       []tls.Certificate{redisClientCert},
+				ClientSessionCache: tls.NewLRUClientSessionCache(100),
+			})
+		},
+	}
+	c := redis.NewFailoverClient(&redisOpts)
+	if err := c.Ping(context.Background()).Err(); err != nil {
```

Comment:
I would prefer if we don't block startup of Zeno on being able to connect to Redis. Redis is really an auxiliary service, and in typical deployments zeno can continue to ingest data and even process most queries without even talking to Redis. If/when Redis comes online, it can start using it for queries.

I think it would be better to just log a warning here, or skip the Ping completely.